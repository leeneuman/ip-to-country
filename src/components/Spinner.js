import React from 'react';

function Spinner({ show = false }) {
	if (!show) return null;

	return <div data-testid="spinner" className="text-secondary spinner-border spinner-border-sm" role="status" />;
}

export default Spinner;
