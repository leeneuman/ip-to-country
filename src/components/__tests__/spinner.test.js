import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Spinner from '../spinner';

afterEach(cleanup);

test('should render spinner component', () => {
	render(<Spinner show />);
	const itemElement = screen.getByTestId('spinner');
	expect(itemElement).toBeInTheDocument();
});

test('matches snapshot', () => {
	const tree = renderer.create(<Spinner show />).toJSON();
	expect(tree).toMatchSnapshot();
});
