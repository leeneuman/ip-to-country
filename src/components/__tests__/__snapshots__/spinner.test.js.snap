// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="text-secondary spinner-border spinner-border-sm"
  data-testid="spinner"
  role="status"
/>
`;
